std/js/lib/Iterator.hx

```diff
@@ -22,11 +22,48 @@
 
 package js.lib;
 
-typedef Iterator<T> = {
+/**
+	abstract over JavaScript iterator objects that enables supports haxe for-in iteration
+**/
+@:forward
+abstract Iterator<T>(IteratorStructure<T>) from IteratorStructure<T> {
```

Comment:
An abstract is used here instead of a class so we can add the `iterator()` field to the ES6 Iterator typedef while still allowing structural unification

Reply:
Missing `to IteratorStructure`?

Reply:
Good spot @RealyUniqueName, just pushed that change